src/opentracing/Recorder.ts

```diff
@@ -34,15 +34,13 @@ class ThundraRecorder {
             for (const listener of this.listeners) {
                 const isCallbackCalled = listener.onSpanStarted(span, options.me,
                     options.callback, options.args, !shouldInvokeCallback);
-                if (shouldInvokeCallback) {
-                    shouldInvokeCallback = !isCallbackCalled;
-                }
-            }
-            if (shouldInvokeCallback) {
-                if (typeof(options.callback) === 'function') {
-                    options.callback.apply(options.me, options.args);
+                if (!isCallbackCalled) {
+                    if (typeof(options.callback) === 'function') {
+                        options.callback.apply(options.me, options.args);
+                    }
                 }
             }
+
         } else if (spanEvent === SpanEvent.SPAN_FINISH) {
```

Comment:
We should do the above changes for the cases where `spanEvent === SpanEvent.SPAN_FINISH` too.